Library/Formula/dmd.rb

```diff
@@ -65,6 +57,27 @@ def install
     end
   end
 
+  # Fix up dmd.conf to point to the correct include paths.
+  def fix_dmd_conf
+    # Homebrew's etc.install method doesn't let you overwrite etc files,
+    # so edit it in place if it's there:
+    conf = etc/"dmd.conf"
+    if conf.exist?
+      inreplace conf, /^DFLAGS=.+$/, "DFLAGS=-I#{include}/d2 -L-L#{lib}"
+    else
+      conf.write <<-EOS.undent
+        [Environment]
+        DFLAGS=-I#{include}/d2 -L-L#{lib}
+        EOS
```

Comment:
Write the conf to somewhere in the `buildpath` in `install` and then use `etc.install` to install it. We only need to do the fixup in post-install, not the write. Also, if you do `opt_include` and `opt_lib` it'll mean that the paths are correct on upgrades.